Chat/Network.cpp

```diff
@@ -0,0 +1,183 @@
+#include "Network.h"
+#include <iostream>
+
+Network::Network() : is_working_(false){}
+
+Network::~Network()
+{
+    Cleanup();
+}
+
+int Network::PrepareNetwork()
+{
+    WSAData wsa_data;
+    int error_check = WSAStartup(WSA_VERSION, &wsa_data);
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    error_check = broadc_socket_.Initialize();
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    error_check = recv_socket_.Initialize();
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    //getting local pc ip
+    int rand_value = rand(), buffer = 0;
+    void *rand_value_send = NULL;
+    buffer = Parcer::PackMessage(PREPARE_MESSAGE, &rand_value, rand_value_send);
+    error_check = broadc_socket_.Send(rand_value_send, buffer);
+    if(error_check == SOCKET_ERROR)
+    {
+        return error_check;
+    }
+    
+    while(true)
+    {
+        RecvStruct recv_struct;
+        recv_socket_.Recv(&recv_struct);
+
+        UnpackedMessage unp_msg = Parcer::UnpackMessage(recv_struct.packet_);
+        if(unp_msg.type_ == PREPARE_MESSAGE ||
```

Comment:
To prevent some compilers strange behavior, please place each condition in brackets.
Statement if( A == B || A == C) may be parsed as
1) (A == B) || (A == C)
2)((A == B) || A) == C)
...
etc.
So, please write as much brackets as you can.